### PR TITLE
layers: Allow more stages in render pass

### PR DESCRIPTION
**Pull request by krOoze**
Allow raytracing stages, `VK_PIPELINE_STAGE_FRAGMENT_DENSITY_PROCESS_BIT_EXT` and `VK_PIPELINE_STAGE_COMMAND_PROCESS_BIT_NVX` in render pass.

### CODE REVIEW

File: layers/stateless_validation.h [hunk 1]
```diff
@@ -825,27 +825,44 @@ class StatelessValidation : public ValidationObject {
...
                                       uint32_t dependency_index, uint32_t subpass, VkPipelineStageFlags stages, const char *vuid,
                                       const char *target) {
-        const VkPipelineStageFlags kGraphicsStages =
-            VK_PIPELINE_STAGE_ALL_GRAPHICS_BIT | VK_PIPELINE_STAGE_TOP_OF_PIPE_BIT | VK_PIPELINE_STAGE_DRAW_INDIRECT_BIT |
-            VK_PIPELINE_STAGE_TASK_SHADER_BIT_NV | VK_PIPELINE_STAGE_MESH_SHADER_BIT_NV | VK_PIPELINE_STAGE_VERTEX_INPUT_BIT |
+        const VkPipelineStageFlags kCommonStages = VK_PIPELINE_STAGE_TOP_OF_PIPE_BIT | VK_PIPELINE_STAGE_BOTTOM_OF_PIPE_BIT;
+        const VkPipelineStageFlags kFramebufferStages =
+            VK_PIPELINE_STAGE_EARLY_FRAGMENT_TESTS_BIT | VK_PIPELINE_STAGE_FRAGMENT_SHADER_BIT |
+            VK_PIPELINE_STAGE_LATE_FRAGMENT_TESTS_BIT | VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT;
+        const VkPipelineStageFlags kPrimitimePipelineStages =
```
Comment (mark-lunarg) [inline layers/stateless_validation.h]: Primetime or Primitive?  Or is it Primitime?

Comment (krOoze) [reply layers/stateless_validation.h]: Why not both? 🤷‍♀ 😄

Comment (mark-lunarg) [reply layers/stateless_validation.h]: 👍

Comment (krOoze) [reply layers/stateless_validation.h]: fixt

File: layers/stateless_validation.h [hunk 2]
```diff
@@ -825,27 +825,44 @@ class StatelessValidation : public ValidationObject {
...
+        const VkPipelineStageFlags kGraphicsStages = VK_PIPELINE_STAGE_ALL_GRAPHICS_BIT | kPrimitiveShadingPipelineStages |
+                                                     kMeshShadingPipelineStages | kFragmentDensityStages |
+                                                     kConditionalRenderingStages | kCommandProcessingPipelineStages;
+
+        const VkPipelineStageFlags kRayTracingPipelineStages = kCommonStages | VK_PIPELINE_STAGE_RAY_TRACING_SHADER_BIT_NV;
+        const VkPipelineStageFlags kRayTracingAccellerationStructOpsStages =
+            kCommonStages | VK_PIPELINE_STAGE_ACCELERATION_STRUCTURE_BUILD_BIT_NV;
+        const VkPipelineStageFlags kRayTracingStages = kCommonStages | kRayTracingPipelineStages |
+                                                       kRayTracingAccellerationStructOpsStages | kFragmentDensityStages |
+                                                       kConditionalRenderingStages | kCommandProcessingPipelineStages;
```
Comment (krOoze) [inline layers/stateless_validation.h]: Technically, I am not even sure this is true, but it better be permissive until it is cleaned up in spec?

**Pull request merged**